File: solutions/c/rna-transcription/1/rna_transcription.c
#include "rna_transcription.h"

#include <string.h>

char* to_rna(const char* dna) {
    char* rna = strdup(dna);
    for (char* c = rna; *c; ++c) {
        *c = *c == 'A'   ? 'U'
             : *c == 'C' ? 'G'
             : *c == 'G' ? 'C'
             : *c == 'T' ? 'A'
                         : *c;
    }
    return rna;
}

File: solutions/go/rna-transcription/1/rna_transcription.go
package strand

func ToRNA(dna string) (rna string) {
	for _, nucleotide := range dna {
		switch nucleotide {
		case 'C':
			rna += "G"
		case 'G':
			rna += "C"
		case 'T':
			rna += "A"
		case 'A':
			rna += "U"
		}
	}
	return
}
